### DISCUSSION TITLE
Do we have any similar property of Azure function's 'batchSize' in KEDA scaler?

### DISCUSSION THREAD

**Discussion by kailashp** [kedacore/keda#4906] [Q&A / Need Help]
We have hosted azure function apps in K8s (Google Cloud Platform). Used [Azure service bus](https://keda.sh/docs/2.10/scalers/azure-service-bus/) scaler in KEDA to scale the pods up/down based on message count in azure service bus queue. This is how our scaler file looks,

![image](https://github.com/kedacore/keda/assets/14346476/f7286762-7c94-4fd6-bf4b-f840b933fddc)

Now the problem is, it is scaling up based `messageCount` property. But not all the pods are processing the messages from queue. Few pods only processing, other pods are just started running but not taking any messages from queue to process.

In [Azure `host.json`](https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-queue?tabs=in-process%2Cextensionv5%2Cextensionv3&pivots=programming-language-csharp#host-json), we have one property called `batchSize`. The value mentioned here will be the number of queue messages that the Functions runtime retrieves simultaneously and processes in parallel.

Do we have any similar property in KEDA scaler? 

Thanks,
Kailash P

**Comment by JorTurFer** [comment]
Hi
What do you want this for? I mean, KEDA doesn't read the messages, it just query to the metadata api yo get the pending count to expose it.  If you want to have all the time your functions at 100%, I guess that you should set the `messageCount `equal to `batchSize`. This wil try to have `batchSize` messages in average for each pod

**Reply by kailashp** [reply]
Hi @JorTurFer,

Thanks for your suggestion. I have also tried setting both `messageCount` and `bacthSize` property as same value. But there is no equivalent property for `batchSize` in KEDA's ScaledObject. Refer the documentation for the same from [here](https://keda.sh/docs/2.10/scalers/azure-service-bus/).

Thanks,
Kailash P

**Reply by JorTurFer** [reply]
@tomkerkhove WDYT?

**Reply by tomkerkhove** [reply]
Can you elaborate how batch size is different? I'm not sure I get the scenario - Sorry
